### PR TITLE
COMMUNITY: Add support for Codacy in Argo CD based deploymet

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Add support for Codacy in Argo CD based deploymet

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - Validated in a staging environment https://cortx-re-vm-6805.pun.seagate.com:31450/applications?showFavorites=false&proj=&sync=&health=&namespace=&cluster=&labels= 

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] Jira ID/ COMMUNITY is prefixed in PR title
      Prefix Syntax - 
        `CORTX-<5 digit JIRA ID>: <PR Title>` - for Seagate Employees
        `COMMUNITY: <PR Title>`  - for open source contributors
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered solutions/dashboard/ElasticsearchKibanaInstallation.md](https://github.com/shailesh-vaidya/cortx-re/blob/argo-codacy/solutions/dashboard/ElasticsearchKibanaInstallation.md)
[View rendered solutions/dashboard/README.md](https://github.com/shailesh-vaidya/cortx-re/blob/argo-codacy/solutions/dashboard/README.md)
[View rendered solutions/dashboard/deployment/README.md](https://github.com/shailesh-vaidya/cortx-re/blob/argo-codacy/solutions/dashboard/deployment/README.md)

### CODE REVIEW

**Review by Sanket292001 — APPROVED** [review]

File: solutions/dashboard/deployment/README.md [hunk 1]
```diff
@@ -42,6 +42,17 @@ Username: admin
 Password:
 'admin:login' logged in successfully
 ```
+- Create ElasticSearch Application using CLI
+```
+argocd app create elasticsearch \
+--repo https://github.com/shailesh-vaidya/cortx-re \
```
Comment (gauravchaudhari02) [inline solutions/dashboard/deployment/README.md:48]: is it going to be your fork?

File: solutions/dashboard/deployment/dashboard/codacy.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+apiVersion: batch/v1
```
Comment (gauravchaudhari02) [inline solutions/dashboard/deployment/dashboard/codacy.yaml:1]: looks like these are changes done by @Sanket292001 right?

File: solutions/dashboard/deployment/README.md (hunk 1 above)
Comment (shailesh-vaidya) [reply solutions/dashboard/deployment/README.md:48]: Yes. Till we fully migrate to Argo CD for deployment

File: solutions/dashboard/deployment/dashboard/codacy.yaml (hunk 2 above)
Comment (shailesh-vaidya) [reply solutions/dashboard/deployment/dashboard/codacy.yaml:1]: @Sanket292001 Added code in https://github.com/Seagate/cortx-re/tree/dashboard/solutions/dashboard/config/codacy which is not tested with Argo CD. I have added code here with Argo CD changes. 
We will be removing https://github.com/Seagate/cortx-re/tree/dashboard/solutions/dashboard/config eventually.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**